lib/google_scraping/dashboard/keywords.ex

```diff
@@ -0,0 +1,59 @@
+defmodule GoogleScraping.Dashboard.Keywords do
+  @moduledoc """
+  The Keywords context.
+  """
+
+  import Ecto.Query, warn: false
+
+  alias Ecto.Multi
+  alias GoogleScraping.Dashboard.Queries.KeywordQuery
+  alias GoogleScraping.Dashboard.Schemas.Keyword
+  alias GoogleScraping.Repo
+
+  @doc """
+  Returns the list of keywords associated with the given user.
+
+  ## Examples
+
+      iex> list_keywords(user_id)
+      [%Keyword{}, ...]
+
+  """
+  def list_keywords(user_id), do: Repo.all(KeywordQuery.user_keywords_query(user_id))
+
+  @doc """
+  Creates a keyword.
+
+  ## Examples
+
+      iex> create_keyword(%{field: value})
+      {:ok, %Keyword{}}
+
+      iex> create_keyword(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_keyword(attrs \\ %{}) do
+    %Keyword{}
+    |> Keyword.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Creates keywords list.
+  """
+  def create_keyword_list(keyword_list, user) do
+    Enum.each(keyword_list, fn keyword_name ->
+      keyword_params = %{
+        user_id: user.id,
+        name: keyword_name,
+        status: :new
+      }
+
+      # TODO: start background jobs for each created keyword
+      Multi.new()
+      |> Multi.run(:keyword, fn _, _ -> create_keyword(keyword_params) end)
+      |> Repo.transaction()
```

Comment:
Hi, I think this case would still have partially processed data since we're looping through the keyword list and committing each transaction per loop, right? 🤔 

I also think we should have a test for this case as well to check that it does rollback. 
I'm fine with just creating the keyword for now and we can implement the rollback feature a separate PR or when you're working on the background jobs. What do you think @Nihisil? 😄

Reply:
You're right, I misread this code block. Let's leave it for future PRs, I need to investigate how properly work with transactions in Ecto. Thanks.

Reply:
Then shall we remove it for now and just leave it as it was before @Nihisil 👍 

```suggestion
      create_keyword(keyword_params)
```

Reply:
Done